Formula/subversion.rb

```diff
@@ -235,21 +233,14 @@ def caveats
     system "#{bin}/svnadmin", "create", "test"
     system "#{bin}/svnadmin", "verify", "test"
 
-    if Hardware::CPU.intel?
-      platform = "darwin-thread-multi-2level"
-      on_linux do
-        platform = "x86_64-linux-thread-multi"
-      end
-
-      perl = "/usr/bin/perl"
-      on_linux do
-        perl = "#{Formula["perl"].opt_bin}/perl"
-      end
```

Comment:
Let's keep this. I think you are correct that we want to make sure that the `perl` used in the test should be the one used at build time. That's exactly what specifying system Perl on macOS and Homebrew Perl on Linux does, because that's exactly what is used at install time.

`brew` filters the `PATH` env for the build but doesn't during the `test`, which is why we need to specify Perl this way in the test but not at install.

Reply:
Btw, for any further changes, please squash your commits together with the first one. Thanks!

Reply:
Reverted and squashed!  I appreciate the input, thanks; I wasn't aware of this build-time path filtering.  Do you know where I can read more about that, or what could potentially cause it to be overridden?  I ask because when I have brew perl installed already and then `brew install --build-from-source subversion`, the perl bindings are being put in a directory containing the brew perl's version number and not the system perl's.  This is causing the test to fail, as it uses the system perl's version number when building the `PERL5LIB` string.

I'm reproducing this readily as follows:
- `brew install perl`
- `which perl` outputs `/usr/local/bin/perl` (on my Intel machine)
- `/usr/bin/perl --version` shows 5.30.2
- `perl --version` shows 5.34.0
- `brew install --build-from-source subversion`
- `ls /usr/local/opt/subversion/lib/perl5/site_perl` outputs 5.34.0
- `ls /usr/local/Cellar/subversion/1.14.1_4/lib/perl5/site_perl` outputs 5.34.0
- `brew test subversion` fails because `Can't locate SVN/Client.pm in @INC`; `@INC` contains:
  - /usr/local/Cellar/subversion/1.14.1_4/lib/perl5/site_perl/5.30.2/darwin-thread-multi-2level
  - /Library/Perl/5.30/darwin-thread-multi-2level
  - /Library/Perl/5.30
  - /Network/Library/Perl/5.30/darwin-thread-multi-2level
  - /Network/Library/Perl/5.30
  - /Library/Perl/Updates/5.30.2
  - /System/Library/Perl/5.30/darwin-thread-multi-2level
  - /System/Library/Perl/5.30
  - /System/Library/Perl/Extras/5.30/darwin-thread-multi-2level
  - /System/Library/Perl/Extras/5.30
- The first of the above paths doesn't exist because the bindings were installed for 5.34.0 and not 5.30.2.

Reply:
Hm, that's odd. In any case, it should be happening here: https://github.com/Homebrew/brew/blob/12d5e70b8732c7f9b46b18fcdf81c9f68a129cd6/Library/Homebrew/extend/ENV/super.rb#L67

Reply:
Thank you for the info.

I'll keep looking into this issue with brew perl and the tests, as I'd like to figure out what's going on there.  However, I don't think it needs to hold up the PR-- the behavior is pre-existing to my change, for Intel Macs at least.

Reply:
I added a change that addresses the brew perl problem.  While `PATH` _is_ being filtered, for some reason Homebrew's `bin` folder is still ending up in it at the point the `install` function runs.  I'm working around that problem by explicitly using the system perl on macOS, following the examples of `krb5`, `ruby`, `sqlite`, and `zlib` already doing the same thing in this function.